src/lib/message/L2ToL1Message.ts

```diff
@@ -66,35 +65,36 @@ export enum L2ToL1MessageStatus {
 export type L2ToL1MessageReaderOrWriter<T extends SignerOrProvider> =
   T extends Provider ? L2ToL1MessageReader : L2ToL1MessageWriter
 
+export type L2ToL1TransactionEvent = {
+  caller: string
+  destination: string
+  hash: BigNumber
+  position: BigNumber
+  indexInBatch?: BigNumber
+  arbBlockNum: BigNumber
+  ethBlockNum: BigNumber
+  timestamp: BigNumber
+  callvalue: BigNumber
+  data: string
+}
+
 /**
  * Base functionality for L2->L1 messages
  */
 export class L2ToL1Message {
   public static fromEvent<T extends SignerOrProvider>(
     l1SignerOrProvider: T,
-    event: L2ToL1TransactionEvent['args'],
-    outboxAddress: string,
-    isNitro: boolean
+    event: L2ToL1TransactionEvent,
+    outboxAddress: string
```

Comment:
I think its cleaner if we get the l2SignerOrProvider and use that to get the outboxAddress from the networks object
Users already setup the networks object with that information, so it would be cool if we didn't require it again in public functions

Reply:
same as on classic sdk, so lets keep for now